### PR TITLE
Delegation and baking support in the mobile wallet library

### PR DESCRIPTION
**Pull request by td202**
## Purpose

Add support for delegation and baker transactions to the mobile wallet library.

## Changes

- Functions for configuring baker and delegation are added together with a function for generating baker keys. 

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [ ] I accept the above linked CLA.

### CODE REVIEW

File: rust-bins/wallet-notes/README.md [hunk 1]
```diff
@@ -152,6 +152,97 @@ The returned value is a JSON object with the following fields:
 An example input to this request is in the file [create_transfer-input.json](files/create_transfer-input.json).
 An example output to this request is in the file [create_transfer-output.json](files/create_transfer-output.json).
 
+## create_configure_delegation_transaction
+
+Semantics: Creates a `configure delegation`-transaction with the provided values.
+
+This function takes as input a NUL-terminated UTF8-encoded string. The string
+must be a valid JSON object with fields
```
[the remark this answers is not in the document]
Comment (eb-concordium) [reply rust-bins/wallet-notes/README.md:160]: Done.

File: rust-bins/wallet-notes/README.md [hunk 2]
```diff
@@ -152,6 +152,97 @@ The returned value is a JSON object with the following fields:
...
+
+- `"keys"` ... mapping with the keys of the sender account.
+
+- `"energy"` ... max energy wanted for the transfer.
+
+- `"capital"` ... string containing the amount to be staked.
+
+- `"restakeEarnings"` ... bool indicating whether earnings should be restaked
+
+- `"delegationTarget"` ... JSON indicating either delegation to the L-pool or to a baker pool.
```
[the remark this answers is not in the document]
Comment (eb-concordium) [reply rust-bins/wallet-notes/README.md:176]: Done.

File: rust-bins/wallet-notes/README.md [hunk 3]
```diff
@@ -152,6 +152,97 @@ The returned value is a JSON object with the following fields:
...
+
+- `"openStatus"` ... whether the pool is closed, open for delegation or closed for new delegators.
+
+- `"transactionFeeCommission"` ... number indicating the transaction fee commission in parts per hundred thousand.
+
+- `"bakingRewardCommission"` ... number indicating the baking reward commission in parts per hundred thousand.
+
+- `"finalizationRewardCommission"` ... number indicating the finalization reward commission in parts per hundred thousand.
+
+- `"bakerKeys"` ... the baker keys.
```
[the remark this answers is not in the document]
Comment (eb-concordium) [reply rust-bins/wallet-notes/README.md:217]: Done

File: rust-bins/wallet-notes/example.c [hunk 4]
```diff
@@ -139,6 +148,12 @@ int main(int argc, char *argv[]) {
       out = combine_encrypted_amounts(argv[2], argv[3], &flag);
       printf("%s\n", out);
       return (int)flag;
+    } else if (strcmp(argv[1], "generate-baker-keys") == 0) {
+      uint8_t flag = 1;
+      char *out;
+      out = generate_baker_keys(&flag);
+      printf("%s\n", out);
+      return (int)flag;
```
Comment (td202) [inline rust-bins/wallet-notes/example.c:156]: Why not use `return printStr(out, flag);`?

File: rust-bins/wallet-notes/README.md [hunk 5]
```diff
@@ -152,6 +152,119 @@ The returned value is a JSON object with the following fields:
...
+- `"nonce"` ... nonce of the sender account.
+
+- `"keys"` ... mapping with the keys of the sender account.
+
+- `"energy"` ... max energy wanted for the transfer.
+
+The following fields are optional:
+
+- `"capital"` ... string containing the amount to be staked.
+
```
Comment (td202) [inline rust-bins/wallet-notes/README.md:227]: Missing: `"restakeEarnings"` and `"url"`. The length limit on the url should be stated.

File: mobile_wallet/src/lib.rs [hunk 6]
```diff
@@ -188,6 +209,235 @@ fn create_transfer_aux(input: &str) -> anyhow::Result<String> {
...
+        Some(m) => Some(from_value(m.clone())?),
+        None => None,
+    };
+
+    let maybe_openstatus: Option<OpenStatus> = match v.get("openStatus") {
+        Some(m) => Some(from_value(m.clone())?),
+        None => None,
+    };
+
+    let maybe_url: Option<UrlText> = match v.get("url") {
```
Comment (td202) [inline mobile_wallet/src/lib.rs:312]: For consistency, I think this should be renamed to `"metadataUrl"`.

File: rust-bins/wallet-notes/README.md (hunk 5 above)
Comment (eb-concordium) [reply rust-bins/wallet-notes/README.md:227]: Done.

File: rust-bins/wallet-notes/example.c (hunk 4 above)
Comment (eb-concordium) [reply rust-bins/wallet-notes/example.c]: Using `return printStr(out, flag);` now.

File: mobile_wallet/src/lib.rs [hunk 7]
```diff
@@ -188,6 +209,235 @@ fn create_transfer_aux(input: &str) -> anyhow::Result<String> {
     Ok(to_string(&response)?)
 }
 
+fn create_configure_delegation_transaction_aux(input: &str) -> anyhow::Result<String> {
+    let v: Value = from_str(input)?;
+
+    let ctx: TransferContext = from_value(v.clone())?;
+
+    let maybe_capital: Option<Amount> = match v.get("capital") {
```
[the remark this answers is not in the document]
Comment (eb-concordium) [reply mobile_wallet/src/lib.rs]: I refactored the code a little bit.

**Review by mkmks — APPROVED** [review]

**Pull request merged**